server.py

```diff
@@ -270,13 +271,24 @@ def asyncFinishThread(server):
 		app.logger.warning("Unable to get address info for %s." % (server["address"],))
 		return
 
+	addresses = set(data[4][0] for data in info)
 	if checkAddress:
-		addresses = set(data[4][0] for data in info)
 		if not server["ip"] in addresses:
 			app.logger.warning("Invalid IP %s for address %s (address valid for %s)."
 					% (server["ip"], server["address"], addresses))
 			return
 
+	reader = geolite2.reader()
+	for addr in addresses:
+		try:
+			geo = reader.get(addr)
+			server["geo_continent"] = geo["continent"]["code"]
+		except:
+			pass
+
+	if not server["geo_continent"]:
```

Comment:
this will actually raise an exception if the key does not exist
you probably want `if "geo_continent" not in server:`